src/Explorer/Tree/Collection.ts

```diff
@@ -526,6 +527,51 @@ export default class Collection implements ViewModels.Collection {
     }
   };
 
+  public onMongoDBSchemaClick = () => {
+    this.container.selectedNode(this);
+    this.selectedSubnodeKind(ViewModels.CollectionTabKind.MongoSchema);
+    TelemetryProcessor.trace(Action.SelectItem, ActionModifiers.Mark, {
+      description: "Mongo Schema node",
+      databaseName: this.databaseId,
+      collectionName: this.id(),
+      dataExplorerArea: Constants.Areas.ResourceTree,
+    });
+
+    const mongoSchemaTabs: MongoSchemaTab[] = this.container.tabsManager.getTabs(
+      ViewModels.CollectionTabKind.MongoSchema,
+      (tab) => tab.collection && tab.collection.databaseId === this.databaseId && tab.collection.id() === this.id()
+    ) as MongoSchemaTab[];
+    let mongoSchemaTab: MongoSchemaTab = mongoSchemaTabs && mongoSchemaTabs[0];
+
+    if (mongoSchemaTab) {
+      this.container.tabsManager.activateTab(mongoSchemaTab);
+    } else {
+      const startKey: number = TelemetryProcessor.traceStart(Action.Tab, {
+        databaseName: this.databaseId,
+        collectionName: this.id(),
+        dataExplorerArea: Constants.Areas.Tab,
+        tabTitle: "Schema",
+      });
+      this.documentIds([]);
+
+      mongoSchemaTab = new MongoSchemaTab({
+        account: userContext.databaseAccount,
+        masterKey: userContext.masterKey || "",
+        container: this.container,
+        tabKind: ViewModels.CollectionTabKind.MongoSchema,
+        title: "Schema",
+        tabPath: "",
+        collection: this,
+        node: this,
+        hashLocation: `${Constants.HashRoutePrefixes.collectionsWithIds(this.databaseId, this.id())}/mongoSchema`,
+        isActive: ko.observable(false),
+        onLoadStartKey: startKey,
+        onUpdateTabsButtons: this.container.onUpdateTabsButtons,
+      });
+      this.container.tabsManager.activateNewTab(mongoSchemaTab);
+    }
```

Comment:
I realize this is just an established pattern, but you can actually avoid using `let` and the `as MongoSchemaTab[]` here and keep things type-safe, like so:

```suggestion
    for (const tab of this.container.tabsManager.openedTabs()) {
      if (
        tab instanceof MongoSchemaTab &&
        tab.collection?.databaseId === this.databaseId &&
        tab.collection?.id() === this.id()
      ) {
        return this.container.tabsManager.activateTab(tab);
      }
    }

    const startKey = TelemetryProcessor.traceStart(Action.Tab, {
      databaseName: this.databaseId,
      collectionName: this.id(),
      dataExplorerArea: Constants.Areas.Tab,
      tabTitle: "Schema",
    });
    this.documentIds([]);

    this.container.tabsManager.activateNewTab(
      new MongoSchemaTab({
        account: userContext.databaseAccount,
        masterKey: userContext.masterKey || "",
        container: this.container,
        tabKind: ViewModels.CollectionTabKind.MongoSchema,
        title: "Schema",
        tabPath: "",
        collection: this,
        node: this,
        hashLocation: `${Constants.HashRoutePrefixes.collectionsWithIds(this.databaseId, this.id())}/mongoSchema`,
        isActive: ko.observable(false),
        onLoadStartKey: startKey,
        onUpdateTabsButtons: this.container.onUpdateTabsButtons,
      })
    );
```